File: JavaScript/Fase3.js
/**
 * Exercici:
 * L’exercici consisteix en mostrar per consola quantes vegades apareix cada lletra del teu nom i cognoms amb diferents variants (fases).
 * FASE 3
 * ● Emmagatzemar en un Map tant les lletres de la llista com el nombre de vegades que apareixen.
 * 
 * @author Enric Comes
 *
 */

 //Variables global que es reutilitza en els bucles
 var comptador = 0

 //Array del nom
 var nom = ["E","n","r","i","c"]

 //Map per emmagatzemar nom i nombre
 var persona = new Map()

 //Bucle per recorrer el nom
 for(var i = 0; i < nom.length ; i++){

    lletra = nom[i] //S'emmagatzema la lletra actual

    //Bucle per a obtenir el nombre de vegades que 
    //apareix la lletra actual
    for(var j = 0; j < nom.length ; j++){
        if(nom[j] == nom[i]) comptador++
    }

    //S'inclou al Map la lletra actual 
    //amb les vegades que surt
    persona.set(nom[i] , comptador)
    comptador = 0 //reinicia comptador per a la següent lletra

 }

 //Per a comprovar que es guarden les lletres de la llista
 //i les vegades que apareixen
 console.log(persona)

File: src/lletresRepetides/Fase3.java
package lletresRepetides;

import java.util.*;

/**
 * Exercici:
 * L’exercici consisteix en mostrar per consola quantes vegades apareix cada lletra del teu nom i cognoms amb diferents variants (fases).
 * FASE 3
 * ● Emmagatzemar en un Map tant les lletres de la llista com el nombre de vegades que apareixen.
 * 
 * @author Enric Comes
 *
 */

public class Fase3 {

	public static void main(String[] args) {
		
		//Variable global que es reutilitza en els bucle
		int comptador = 0;
		
		//Llista de Characters per al nom
		List<Character> nom = new ArrayList<>();
		
		//El nom que es guardarà a la llista
		nom.add('E');
		nom.add('n');
		nom.add('r');
		nom.add('i');
		nom.add('c');
		
		//LinkedHashMap per emmagatzemar lletra i nombre per ordre d'inclusió
		LinkedHashMap< String, String> persona = new LinkedHashMap<String, String>();				
		
		//Bucle per recorrer el nom
		for (int i=0; i<nom.size(); i++) {	
			
			//Bucle per a obtenir el nombre de vegades que apareix la lletra actual
			for(int j=0; j<nom.size(); j++) {
				if(nom.get(j) == nom.get(i)) comptador++;
			}
			
			//S'inclou al Map la lletra actual i les vegades que hi surt
			persona.put(String.valueOf(nom.get(i)), Integer.toString(comptador));
			comptador=0; //reinici del comptador per a la següent lletra
			
		}
		
		//Per a comprovar que es guarden les lletres de la llista i el nombre de vegades que apareixen
		System.out.println(persona);
	}

}
